be/src/io/input_stream.h

```diff
@@ -21,7 +21,7 @@ class InputStream : public Readable {
     virtual Status skip(int64_t count) = 0;
 
     // Returns true if InputStream support `peak()`
-    virtual bool allows_peak() const = 0;
+    virtual bool allows_peak() const { return false; }
```

Comment:
is there any upper class will call this method? maybe we don't need `peak`?

Reply:
Not yet, but there will be.